### PR TITLE
Don't print extra metadata in stdout/stderr, and implement proc_exit

### PR DESCRIPTION
**Pull request by sunfishcode**
Import wasi-logging twice, once as stdout and once as stderr, and change the host implementation to just pass the string through to the host stdout/stderr as appropriate.

Also, add an API to implement proc_exit with, and implement it.

### CODE REVIEW

**Comment by alexcrichton** [comment]
> Import wasi-logging twice

FWIW this won't work in planned updates for WIT/components/etc because `wasi-logging`, as a single interface, will have a single URL and imports must all have unique URLs. This is allowed today since this restriction is new, but is it possible to recast this dependency without the usage of importing the same interface?

(cc @lukewagner as well)

**Comment by lukewagner** [comment]
Just to expand on this a bit, to import a single logical interface twice you have single URL for an instance that contains the multiple interfaces without URLs, only kebab-names, such that kebab-name uniqueness is sufficient.  E.g.:
```
interface "wasi:logging/backends" {
  stdout: ...
  stderr: ...
}
```
This does raise the question of what goes in the `...` in this example.  I can always write `stdout: interface { ... }` to produce an inline interface with no URL, but what if I want to still reference `wasi:logging/logger` out-of-line definition but *not* embed the URL?  This is a syntax question we'll have to answer and I don't have an obvious answer atm.

That all being said, could we have a single `wasi:logging/backend` interface that takes an `enum { info, error }` where `info` maps to `stdout` and `error` maps to `stderr`?

**Comment by sunfishcode** [comment]
I've remove the stdout/stderr parts here, so this PR is now just adding the exit function. I'll submit a separate PR with the stdout/stderr changes.

File: host/src/exit.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+use crate::{wasi_exit, WasiCtx};
+
+#[async_trait::async_trait]
+impl wasi_exit::WasiExit for WasiCtx {
+    async fn exit(&mut self, status: Result<(), ()>) -> anyhow::Result<()> {
+        let status = match status {
+            Ok(()) => 0,
+            Err(()) => 1,
+        };
+        std::process::exit(status)
```
Comment (pchickey) [inline host/src/exit.rs:12]: Process exit is inappropriate - instead, we need to return a trap here. We shoul

File: wit/wasi.wit [hunk 2]
```diff
@@ -3,6 +3,11 @@ interface command {
   command: func(stdin: descriptor, stdout: descriptor, args: list<string>)
 }
 
+interface wasi-exit {
+  /// Exit the curerent instance and any linked instances.
```
Comment (pchickey) [inline wit/wasi.wit:7]:
```suggestion
  /// Exit the current instance and any linked instances.
```

**Review by pchickey — APPROVED** [review]

**Pull request merged**